drivers/wifi/nrf700x/src/wifi_mgmt.c

```diff
@@ -791,19 +797,27 @@ int nrf_wifi_mode(const struct device *dev,
 
 	} else {
 		mode->mode = def_dev_ctx->vif_ctx[vif_ctx_zep->vif_idx]->mode;
-
-#if defined(CONFIG_NRF700X_RAW_DATA_TX) || defined(CONFIG_NRF700X_RAW_DATA_RX)
+		/**
+		 * This is a work-around to handle current UMAC mode handling.
+		 * This might be removed in future versions when UMAC has more space.
+		 */
+#ifdef CONFIG_NRF700X_RAW_DATA_TX
 		if (def_dev_ctx->vif_ctx[vif_ctx_zep->vif_idx]->txinjection_mode == true) {
 			mode->mode ^= NRF_WIFI_TX_INJECTION_MODE;
 		}
-#endif
+#endif /* CONFIG_NRF700X_RAW_DATA_TX */
+#ifdef CONFIG_NRF700X_PROMISC_DATA_RX
+		if (def_dev_ctx->vif_ctx[vif_ctx_zep->vif_idx]->promisc_mode == true) {
+			mode->mode ^= NRF_WIFI_PROMISCUOUS_MODE;
```

Comment:
This is the only place where the define is used as standalone, so, do we really need a separate define for promiscuous ? Can't we use RAW_DATA_RX for both and rely on `mode` runtime setting?

Reply:
It has other checks specifically for Promiscuous mode. One of the ask initially was to keep different implementations completely code configurable during build. It was as per that discussion that Monitor, Promiscuous and TX Injection modes are kept separate.

Reply:
> It has other checks specifically for Promiscuous mode

Ok, in the driver?

> It was as per that discussion that Monitor, Promiscuous and TX Injection modes are kept separate.

Anyways, I see a lot of duplication b/w them, so, we should re-discuss, but we can proceed with the PR.